plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/BaseTransactionsTable.java

```diff
@@ -140,7 +137,7 @@ public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHand
         TrinoFileSystem fileSystem = fileSystemFactory.create(session);
         PageListBuilder pagesBuilder = PageListBuilder.forTable(tableMetadata);
         try {
-            List<Transaction> transactions = loadNewTailBackward(fileSystem, table.location(), startVersionExclusive, endVersionInclusive.get()).reversed();
+            List<Transaction> transactions = loadNewTail(fileSystem, table.location(), startVersionExclusive, endVersionInclusive, DataSize.ofBytes(0)).getTransactions();
```

Comment:
It looks like `endVersionInclusive` will never be empty, so we should retain that clarity and unwrap the optional like before

Reply:
added a check for it